File: app/src/main/java/h_mal/appttude/com/driver/Global/ViewController.java
package h_mal.appttude.com.driver.Global;

import android.app.Activity;
import android.view.View;

public class ViewController {

    private Activity activity;

    public ViewController(Activity activity) {
        this.activity = activity;
    }

    public void progress(int vis){
        if (activity instanceof ViewControllerInterface){
            ((ViewControllerInterface) activity).progressVisibility(vis);
        }
    }

    public void reloadDrawer(){
        if (activity instanceof ViewControllerInterface){
            ((ViewControllerInterface) activity).updateDrawer();
        }
    }

    public interface ViewControllerInterface {

        void progressVisibility(int vis);

        void updateDrawer();
    }
}

File: app/src/main/java/h_mal/appttude/com/driver/Global/ViewController.kt
package h_mal.appttude.com.driver.Global

import android.app.Activity


class ViewController constructor(private val activity: Activity?) {
    fun progress(vis: Int) {
        if (activity is ViewControllerInterface) {
            (activity as ViewControllerInterface).progressVisibility(vis)
        }
    }

    fun reloadDrawer() {
        if (activity is ViewControllerInterface) {
            (activity as ViewControllerInterface).updateDrawer()
        }
    }

    open interface ViewControllerInterface {
        fun progressVisibility(vis: Int)
        fun updateDrawer()
    }
}